core/src/main/java/org/mskcc/cbio/portal/dao/DaoCancerStudy.java

```diff
@@ -577,6 +575,10 @@ public static void deleteCancerStudy(int internalCancerStudyId) throws DaoExcept
         ResultSet rs = null;
         try {
             con = JdbcUtil.getDbConnection(DaoCancerStudy.class);
+            // check whether should delete generic assay meta
+            if (DaoGenericAssay.geneticEntitiesOnlyExistInSingleStudy(internalCancerStudyId)) {
```

Comment:
I'm sorry I wasn't clearer here - we should move this statement up before allocating a db connection to minimize the number of connections in use at any one time.